projects/kit/components/range/range.style.less

```diff
@@ -0,0 +1,160 @@
+@import 'taiga-ui-local';
+
+@track-height: 0.125rem;
+@extra-click-area-space: 0.4375rem;
+
+:host {
+    position: relative;
+    display: block;
+    height: @track-height;
+    border-radius: var(--tui-radius-m);
+    background: var(--tui-base-03);
+    cursor: pointer;
+    outline: none;
+    margin: @extra-click-area-space 0;
+
+    &:after {
+        content: '';
+        position: absolute;
+        top: -@extra-click-area-space;
+        bottom: -@extra-click-area-space;
+        width: 100%;
+    }
+
+    &._disabled {
+        opacity: var(--tui-disabled-opacity);
+        cursor: auto;
+    }
+}
+
+.track(@thumb-width) {
+    position: relative;
+    margin: 0 @thumb-width / 2;
+    height: 100%;
+
+    /* Filled selected range */
+    &:after {
+        content: '';
+        position: absolute;
+        top: 0;
+        left: var(--left);
+        right: var(--right);
+        height: 100%;
+        background: var(--tui-primary);
+        margin: 0 -@thumb-width / 2;
+    }
+}
+
+.track {
+    :host[data-size='s'] & {
+        .track(@thumb-diameters[@s]);
+    }
+
+    :host[data-size='m'] & {
+        .track(@thumb-diameters[@m]);
+    }
+}
+
+.ignore-track-pointer-events() {
+    pointer-events: none;
+
+    &::-webkit-slider-thumb {
+        pointer-events: all;
+    }
+
+    &::-moz-range-thumb {
+        pointer-events: all;
+    }
+}
+
+.remove-track-background() {
+    /* Artificially increased specificity */
+    input[type='range']& {
+        &::-webkit-slider-runnable-track {
+            background: transparent;
+        }
+
+        &::-moz-range-track {
+            background: transparent;
+        }
+
+        &::-moz-range-progress {
+            background: transparent;
+        }
+    }
+}
+
+.thumb {
+    .ignore-track-pointer-events();
+    .remove-track-background();
+    position: absolute;
+    top: @track-height / 2;
+    left: 0;
+    right: 0;
+    z-index: 1;
```

Comment:
What is it for?

Reply:
Thumb is filled by semi-transparent color during hover.
It will show what happens under it otherwise.